serpextract/serpextract.py

```diff
@@ -432,15 +432,15 @@ def parse(self, url_parts):
 
                 # Now we have to check for a tricky case where it is a SERP but
                 # there are no keywords
-                if keyword is None and (u'&{}='.format(extractor) in query or
-                                        u'?{}='.format(extractor) in query):
+                if not keyword and (u'&{}='.format(extractor) in original_query or
```

Comment:
the query object will never have the '&q=' or '?q=' string in it, since it's a dict:

```
(Pdb) query
{u'usg': [u'AFQjCNG4c9vrYPeffmVxckMAfRj51PMlpA'], u'rct': [u'j'], u'sqi': [u'2'], u'cd': [u'5'], u'q': [u''], u'source': [u'web'], u'ved': [u'0ahUKEwipjbfT-aPSAhUnCcAKHTbHALEQFgg7MAQ'], u'esrc': [u's'], u'url': [u'https://www.thesun.co.uk/tvandshowbiz/2735307/antonio-banderas-rushed-to-surrey-hospital-after-suffering-agonising-chest-pains-during-workout/'], u'sa': [u't'], u'bvm': [u'bv.147448319,d.ZGg']}
```

original_query does have the url string with the ?q= syntax in it:

```
(Pdb) original_query
u'sa=t&rct=j&q=&esrc=s&source=web&cd=5&sqi=2&ved=0ahUKEwipjbfT-aPSAhUnCcAKHTbHALEQFgg7MAQ&url=https%3A%2F%2Fwww.thesun.co.uk%2Ftvandshowbiz%2F2735307%2Fantonio-banderas-rushed-to-surrey-hospital-after-suffering-agonising-chest-pains-during-workout%2F&usg=AFQjCNG4c9vrYPeffmVxckMAfRj51PMlpA&bvm=bv.147448319,d.ZGg'
```

so if we're looking to see if that substring exists, it'll never match in query but it will match in original_query